### PR TITLE
Java client: Add CI for publishing to Maven Central

### PR DESCRIPTION
**Pull request by batiati**
This PR adds a Maven Central publish step, replacing the GitHub registry.
The version number must be unique and incremental, it's composed by a fixed version `0.0.1` dash `git commit count`.

Also adds badges for `javadocs` and `maven-central` on the Java client's ReadMe.md:

[javadoc](https://javadoc.io/doc/com.tigerbeetle/tigerbeetle-java)

[maven-central](https://central.sonatype.com/namespace/com.tigerbeetle)

**IMPORTANT¹**: The following SECRETS must be created on this repository before merging this PR:

`MAVEN_CENTRAL_TOKEN`
`MAVEN_CENTRAL_USERNAME`
`MAVEN_GPG_SECRET_KEY`
`MAVEN_GPG_SECRET_KEY_PASSWORD`

Please contact me on private to obtain the values to be configured.
More info about setting up github secrets can be found [here](https://docs.github.com/en/actions/security-guides/encrypted-secrets)

**IMPORTANT²**: We can delete the old [GitHub's registry package](https://github.com/tigerbeetledb/tigerbeetle/packages/1734273) after merging this PR.

## Pre-merge checklist

Performance:

* [X] I am very sure this PR could not affect performance.

### CODE REVIEW

**Comment by eatonphil** [comment]
Could we keep publishing to github as well?

**Comment by batiati** [comment]
> Could we keep publishing to github as well?

Technically we can, but there is little gain by doing so, since Maven Central is the-facto standard and GH's repository requires authentication and is too complicated to setup.

I'd prefer avoid having two registries for the same artifact.

File: .github/workflows/clients-java-build.yml [hunk 1]
```diff
@@ -22,12 +19,18 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
+      with:
```
Comment (cb22) [inline .github/workflows/clients-java-build.yml:22]: Do we need the `fetch-depth` set to 0 here?

File: .github/workflows/clients-java-build.yml [hunk 2]
```diff
@@ -22,12 +19,18 @@ jobs:
...
+        fetch-depth: 0    
     - name: Set up JDK
       uses: actions/setup-java@v3
       with:
         java-version: '11'
         distribution: 'temurin'
-        server-id: github
+        server-id: ossrh 
+        server-username: MAVEN_USERNAME
```
Comment (cb22) [inline .github/workflows/clients-java-build.yml:30]: Are these meant to be these static values and not like eg `${{ secrets.MAVEN_CENTRAL_USERNAME }}` below?

File: .github/workflows/clients-java-build.yml (hunk 1 above)
Comment (batiati) [reply .github/workflows/clients-java-build.yml:22]: We need to auto-increment the version number using the commits count. 
By default `actions/checkout@v3` only fetches the last commit, so without that setting `git rev-list --count HEAD` always returns `1`.

File: .github/workflows/clients-java-build.yml (hunk 2 above)
Comment (batiati) [reply .github/workflows/clients-java-build.yml:30]: Yes, it's the name of the env variable, declared below using `${{ secrets.MAVEN_CENTRAL_USERNAME }}`

**Comment by cb22** [comment]
bors cancel

**Comment by cb22** [comment]
(I was too eager with the merge - let me ask @jorangreef to add the secrets when he gets in, then I'll r+ again)

**Comment by cb22** [comment]
@batiati could you DM said secrets to @jorangreef on Slack?

**Pull request merged**